crates/core/src/node/inner/vm_runner.rs

```diff
@@ -465,8 +484,27 @@ impl VmRunner {
                     tx_index += 1;
                 }
                 Err(e) => {
-                    sh_err!("Error while executing transaction: {e}");
-                    executor.rollback_last_tx().await?;
+                    match &e {
+                        // Validation errors are reported and the execution proceeds
+                        AnvilNodeError::TransactionValidationFailedGasLimit {..}|
+                        AnvilNodeError::TransactionValidationFailedMaxFeePerGasTooLow {..} |
+                        AnvilNodeError::TransactionValidationFailedMaxPriorityFeeGreaterThanMaxFee {..} => {
```

Comment:
Is there a way to group validation errors into a subtype? Otherwise it will be very easy to not update this place when we add extra validation checks

Reply:
Good point. I'll move them to a separate component and wrap here.